src/bodyScrollLock.js

```diff
@@ -230,10 +230,12 @@ export const enableBodyScroll = (targetElement: any): void => {
       documentListenerAdded = false;
     }
   } else if (locks.length === 1 && locks[0].targetElement === targetElement) {
-    restoreOverflowSetting();
-
     locks = [];
   } else {
     locks = locks.filter(lock => lock.targetElement !== targetElement);
   }
+
+  if (!locks.length) {
+    restoreOverflowSetting();
```

Comment:
why do we need to run this for even ios devices?

Reply:
@willcpo I don't tested changes on iOS.
Anyway the same issue can appear on any device, so I assume that this change is required for iOS devices too.